### DISCUSSION TITLE
Is there a way to tell if a package is already cloned and built, without causing it to be activated?

### DISCUSSION THREAD

**Discussion by dabrahams** [raxod502/straight.el#1065] [Q&A]
I'd like to create a wrapper around `use-package` that evaluates the expanded `use-package` form immediately if the package is ready, and otherwise invokes `straight-use-package` asynchronously to get the package built, and evaluates the `use-package` form in the completion handler.  Seems simple enough, if I can figure out whether the package is ready for use.  Is that something I can find out?

Thanks,
Dave

**Comment by raxod502** [comment]
Does passing non-nil arguments for both `no-clone` and `no-build` to `straight-use-package` work for your use case? It's not exactly the same as what you asked for, since the package would be _activated_ if it was already cloned and built, but I think that is what you would want anyway.

**Comment by dabrahams** [comment]
I'm not sure; I currently use `straight-use-package-by-default t` and let calls to `use-package` cause stuff to be installed, so I'm only calling `straight-use-package` explicitly to install `use-package` itself.  How should I change my usage pattern to try this?

Thanks!

**Reply by raxod502** [reply]
The extra arguments aren't supported in the default `use-package` integration because this is an unusual use case. But if you're writing your own `use-package` wrapper you barely benefit from the default integration. Instead of generating a `use-package` form with `:straight`, just invoke `straight-use-package` before `use-package`.
